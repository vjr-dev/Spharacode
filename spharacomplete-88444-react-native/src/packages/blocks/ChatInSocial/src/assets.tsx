// Customizable Area Start
export const back1 = require("../assets/image_background-1.png");
export const back2 = require("../assets/image_background-2.png");
export const image_back = require("../assets/image_back.png");
export const LOGO = require("../assets/LOGO.png");
export const Pin = require("../assets/Pin.png");
export const Delete = require("../assets/Delete.png");
export const Archive = require("../assets/Archive.png");
export const More = require("../assets/More.png");
export const menu = require("../assets/Menu.png");
export const search = require("../assets/search.png");
export const check = require("../assets/check.png");
// Customizable Area End