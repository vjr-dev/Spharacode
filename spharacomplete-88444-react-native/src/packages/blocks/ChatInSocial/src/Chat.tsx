// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";
import {
  ActivityIndicator,
  Dimensions,
  FlatList,
  Image,
  ImageBackground,
  Modal,
  SafeAreaView,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from "react-native";

import Swipeout from "react-native-swipeout";
import { COLORS } from "../../../framework/src/Globals";
import * as IMAGE from "./assets";
import ChatController from "./ChatController";
import { Styles } from "./ChatStyle";

const windowWidth = Dimensions.get("window").width;
const windowHeight = Dimensions.get("window").height;

export default class Chat extends ChatController {
  label = (label: string) => {
    if (label === "pin") {
      return "Pin";
    } else if (label === "delete") {
      return "Delete";
    } else if (label === "archive") {
      return "Archive";
    } else {
      return "More";
    }
  };
  imageSource = (label: string) => {
    if (label === "pin") {
      return IMAGE.Pin;
    } else if (label === "delete") {
      return IMAGE.Delete;
    } else if (label === "archive") {
      return IMAGE.Archive;
    } else {
      return IMAGE.menu;
    }
  };
  component(item: any, label: string) {
    return (
      <TouchableOpacity
        testID="componentButton"
        onPress={() => {
          if (label === "pin") {
            this.pinClick(item);
          } else if (label === "delete") {
            this.DeleteClick(item);
          } else if (label === "archive") {
            this.ArchiveClick(item);
          } else {
            this.MoreClick(item);
          }
        }}
        style={Styles.swipMenuButton}
      >
        <Image
          resizeMode="contain"
          source={this.imageSource(label)}
          style={Styles.swipeMenuImage}
        />
        <Text style={Styles.swipeMenuText}>{this.label(label)}</Text>
      </TouchableOpacity>
    );
  }
  swipeoutOne(item, index) {
    return (
      <Swipeout
        close={this.state.close}
        autoClose={true}
        style={{ marginTop: index == 0 ? 10 : 13 }}
        backgroundColor="trancperant"
        left={[
          {
            component: this.component(item, "pin"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "delete"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "archive"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "more"),
            backgroundColor: COLORS.orangelight,
          },
        ]}
      >
        <TouchableOpacity
          onPress={() => this.onclick()}
          style={Styles.chatPersonList}
        >
          <View style={[Styles.chatPersonImageView, { width: "20%" }]}>
            <TouchableOpacity
              testID="chatItemButton"
              onPress={() => this.onclick()}
              style={Styles.chatPerson}
            >
              {item?.avatar ? (
                <Image
                  resizeMode="cover"
                  style={Styles.chatPersonImage}
                  source={{ uri: item.avatar }}
                />
              ) : (
                <View style={Styles.contactText}>
                  <Text
                    style={{
                      textTransform: "uppercase",
                      color: "black",
                      fontWeight: "bold",
                      fontSize: 18,
                    }}
                  >
                    {item?.name.charAt(0)}
                  </Text>
                </View>
              )}
            </TouchableOpacity>
          </View>
          <TouchableOpacity
            testID="conversationButton"
            onPress={() =>
              this.props.navigation.navigate("ConversationScreen", {
                chatNumber: item.uid,
                mainConversationId: item.conversationId,
                userName: item.name,
                userAvtar: item.avatar,
                userType: 1,
              })
            }
            style={Styles.chatPersonNameView}
          >
            <Text style={Styles.chatPersonName}>{item?.name}</Text>
            <Text style={{ color: COLORS.lightwhite }}>
              {item?.lastMessage}
            </Text>
          </TouchableOpacity>
          <View style={[Styles.chatPersonImageView, { width: "15%" }]}>
            <Text
              style={{
                color: item?.Count == 0 ? COLORS.lightwhite : COLORS.white,
              }}
            >
              {item?.Time}
            </Text>

            {item?.unReadCounts == 0 || isNaN(item?.unReadCounts) ? null : (
              <View style={Styles.chatUnreadMessage}>
                <Text>{item.unReadCounts}</Text>
              </View>
            )}
          </View>
        </TouchableOpacity>
      </Swipeout>
    );
  }

  swipeoutTwo(item, index) {
    return (
      <Swipeout
        close={this.state.close}
        autoClose={true}
        style={{ marginTop: index == 0 ? 10 : 13 }}
        backgroundColor="trancperant"
        left={[
          {
            component: this.component(item, "pin"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "delete"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "archive"),
            backgroundColor: "#363636",
          },
          {
            component: this.component(item, "more"),
            backgroundColor: COLORS.orangelight,
          },
        ]}
      >
        <TouchableOpacity
          // onPress={() => this.onclick()}
          style={Styles.chatPersonList}
        >
          <View style={[Styles.chatPersonImageView, { width: "20%" }]}>
            <TouchableOpacity
              testID="chatItemButton"
              onPress={() => this.onclick()}
              style={Styles.chatPerson}
            >
              {item?.icon ? (
                <Image
                  resizeMode="contain"
                  style={Styles.chatPersonImage}
                  source={{ uri: item.icon }}
                />
              ) : (
                <View style={Styles.contactText}>
                  <Text
                    style={{
                      textTransform: "uppercase",
                      color: "black",
                      fontWeight: "bold",
                      fontSize: 18,
                    }}
                  ></Text>
                </View>
              )}
            </TouchableOpacity>
          </View>
          <TouchableOpacity
            testID="conversationButton"
            onPress={() =>
              this.props.navigation.navigate("ConversationScreen", {
                chatNumber: item.guid,
                mainConversationId: item.conversationId,
                userName: item.name,
                userAvtar: item.icon,
                userType: 2,
              })
            }
            style={Styles.chatPersonNameView}
          >
            <Text style={Styles.chatPersonName}>{item?.name}</Text>
            <Text style={{ color: COLORS.lightwhite }}>
              {item?.lastMessage}
            </Text>
          </TouchableOpacity>
          <View style={[Styles.chatPersonImageView, { width: "15%" }]}>
            <Text
              style={{
                color: item?.Count == 0 ? COLORS.lightwhite : COLORS.white,
              }}
            >
              {item?.Time}
            </Text>
          </View>
        </TouchableOpacity>
      </Swipeout>
    );
  }
  addFlatList() {
    return (
      <>
        <FlatList
          data={
            this.state.Section == 1
              ? this.state.Chat_Data
              : this.state.Group_Data
          }
          testID="conversationList"
          style={{ width: windowWidth, flexGrow: 1 }}
          contentContainerStyle={{ paddingBottom: 500 }}
          showsVerticalScrollIndicator={false}
          keyExtractor={(index): any => index}
          renderItem={({ item, index }: any) => {
            return this.state.Section == 1
              ? this.swipeoutOne(item, index)
              : this.swipeoutTwo(item, index);
          }}
        />
      </>
    );
  }

  ConditionOne() {
    return (
      <>
        {this.state.Loader ? (
          <View style={{ flex: 1 }}>
            <Modal visible={this.state.Loader} transparent={true}>
              <View
                style={{
                  height: windowHeight,
                  width: windowWidth,
                  alignItems: "center",
                  justifyContent: "center",
                }}
              >
                <View
                  style={{
                    height: 100,
                    width: windowWidth - 20,
                    alignSelf: "center",
                    justifyContent: "center",
                    alignItems: "center",
                  }}
                >
                  <ActivityIndicator
                    animating={true}
                    size={"large"}
                    color="#f07233"
                  />
                </View>
              </View>
            </Modal>
          </View>
        ) : (
          <View style={{ flexGrow: 1 }}>
            {this.state.totalUnReadCount > 0 ? (
              <Text style={Styles.unreadMessage}>
                You have {this.state.totalUnReadCount} unread message
              </Text>
            ) : null}
            {this.addFlatList()}
          </View>
        )}

        {this.ConditionTwo()}
      </>
    );
  }

  ConditionTwo() {
    return (
      <>
        {this.state.modal1Visible || this.state.modal2Visible ? (
          <View style={Styles.shadowView} />
        ) : null}
        {this.state.modal1Visible ? (
          <View style={{ flex: 1 }}>
            <View style={Styles.modalView}>
              <View style={Styles.menuView}>
                <TouchableOpacity>
                  <Text style={Styles.menuText}>New Chat</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={Styles.menuText}>Edit Chats</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={Styles.menuText}>New Broadcast</Text>
                </TouchableOpacity>
              </View>
              <TouchableOpacity
                testID="cancelButton"
                onPress={() => this.setState({ modal1Visible: false })}
                style={[Styles.cancelButton, { marginTop: 10 }]}
              >
                <Text style={Styles.cancelText}>CANCEL</Text>
              </TouchableOpacity>
            </View>
          </View>
        ) : null}
        {this.state.modal2Visible ? (
          <View style={{ flex: 1 }}>
            <View style={Styles.modalView}>
              <View style={Styles.menuView}>
                <TouchableOpacity>
                  <Text style={Styles.menuText}>Mute</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={Styles.menuText}>Mark as unread</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={Styles.menuText}>Contact info</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={Styles.menuText}>Clear Chat</Text>
                </TouchableOpacity>
                <View style={Styles.seperator} />
                <TouchableOpacity>
                  <Text style={[Styles.menuText, { color: COLORS.red }]}>
                    Delete Chat
                  </Text>
                </TouchableOpacity>
              </View>
              <TouchableOpacity
                testID="secondModalCancelButton"
                onPress={() => this.setState({ modal2Visible: false })}
                style={[Styles.cancelButton, { marginTop: 15 }]}
              >
                <Text style={Styles.cancelText}>CANCEL</Text>
              </TouchableOpacity>
            </View>
          </View>
        ) : null}
      </>
    );
  }
  render() {
    return (
      <ImageBackground source={IMAGE.back1} style={Styles.container}>
        <ImageBackground source={IMAGE.back2} style={Styles.container}>
          <SafeAreaView>
            <View
              style={[
                Styles.headerView,
                { elevation: this.state.modal1Visible ? 0 : 10 },
              ]}
            >
              <View style={{ width: "16%" }} />
              <View style={Styles.headerCenterView}>
                <TouchableOpacity
                  testID="switchbtnID"
                  onPress={() => this.checkStateUpdate(1)}
                  style={[
                    Styles.switchbutton,
                    {
                      backgroundColor:
                        this.state.Section == 1
                          ? COLORS.orangelight
                          : "transparent",
                    },
                  ]}
                >
                  <Text
                    style={[
                      Styles.switchButtonText,
                      {
                        color:
                          this.state.Section == 1
                            ? COLORS.black
                            : COLORS.lightwhite,
                      },
                    ]}
                  >
                    Chats
                  </Text>
                </TouchableOpacity>
                <TouchableOpacity
                  testID="groupButton"
                  onPress={() => this.checkStateUpdate(2)}
                  style={[
                    Styles.switchbutton,
                    {
                      backgroundColor:
                        this.state.Section == 2
                          ? COLORS.orangelight
                          : "transparent",
                      marginRight: 6,
                    },
                  ]}
                >
                  <Text
                    style={[
                      Styles.switchButtonText,
                      {
                        color:
                          this.state.Section == 2
                            ? COLORS.black
                            : COLORS.lightwhite,
                      },
                    ]}
                  >
                    Groups
                  </Text>
                </TouchableOpacity>
              </View>
              <TouchableOpacity
                testID="backHomeButton"
                onPress={() => this.GoBackToHomeScreen()}
                style={Styles.headerEndView}
              >
                <Image source={IMAGE.LOGO} style={Styles.headerLogo} />
              </TouchableOpacity>
            </View>

            <View style={Styles.searchContainer}>
              <View style={Styles.searchView}>
                <Image source={IMAGE.search} style={Styles.searchImage} />
                <TextInput
                  testID="searchTextInput"
                  placeholder="Search"
                  onChangeText={(e) => this.onSearch(e)}
                  placeholderTextColor="white"
                  style={Styles.searchInput}
                />
              </View>
              <View style={Styles.searchMenuView}>
                <TouchableOpacity
                  testID="menuButton"
                  onPress={() => this.props.navigation.navigate("NewChat")}
                  style={Styles.searchMenuButton}
                >
                  <Image
                    resizeMode="contain"
                    source={IMAGE.menu}
                    style={Styles.serachMenuImage}
                  />
                </TouchableOpacity>
              </View>
            </View>
            {this.ConditionOne()}
          </SafeAreaView>
        </ImageBackground>
      </ImageBackground>
    );
  }
}
// Customizable Area End
