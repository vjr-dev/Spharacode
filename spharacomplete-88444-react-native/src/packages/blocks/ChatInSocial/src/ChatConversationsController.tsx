// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { BlockComponent } from "framework/src/BlockComponent";
import { IBlock } from "framework/src/IBlock";
import { runEngine } from "framework/src/RunEngine";
import { CometChat } from "@cometchat-pro/react-native-chat";
export const configJSON = require("./config");
export interface Props {
  navigation: any;
  id: string;
}
interface S {
  screenFlag:string,
  searchText: string,
  isSwipeoutClose: boolean,
  conversationList: any,
  isLoading: boolean,
  close: boolean,
  isFlatListFooterLoading: boolean,
  isVisibleUserSelectionScreen: boolean,
  unReadCount: number
}
interface SS {
  id: any;
}

export default class ChatConversationsController extends BlockComponent<Props, S, SS> {
  userConversationsRequest: any;
  groupConversationsRequest: any;
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    this.state = {
        screenFlag: props.screenFlag,
        searchText: '',
        isSwipeoutClose:false,
        conversationList:[],
        isLoading: true,
        close:false,
        isFlatListFooterLoading: false,
        isVisibleUserSelectionScreen: false,
        unReadCount:0
     

    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }
  static getDerivedStateFromProps(props, state) {
    if(props.screenFlag !== state.screenFlag){
     
      return{
        screenFlag: props.screenFlag,
        conversationList:[]
      }
    }
  }
  componentDidUpdate(prevProps){
    if(prevProps.screenFlag !== this.state.screenFlag){
      this.setState({isLoading:true})
      this.getAllUnReadMessageCount();
      this.resetConversationRequest();
      this.getConversations();
    }
  }
   componentDidMount(): Promise<void> {
    this.getAllUnReadMessageCount();
      this.resetConversationRequest();
      this.getConversations();
  }
  resetConversationRequest(){
    this.setState({conversationList:[]})
    if(this.state.screenFlag === 'chats'){
        this.userConversationsRequest = new CometChat.ConversationsRequestBuilder()
        .setLimit(30)
        .setConversationType('user')
        //.setTags(['pin'])
        .withTags(true)
        .build();
     
    }else{
      this.groupConversationsRequest = new CometChat.ConversationsRequestBuilder()
      .setLimit(30)
     // .setTags(['pin'])
     .withTags(true)
      .setConversationType('group')
      .build();
    }
  }
  getConversations(){
    if(this.state.screenFlag === 'chats'){
      this.fetchUserConversation();
    }else{
      this.fetchGroupConversation();
    }
       
  }
  fetchUserConversation(){
    this.userConversationsRequest.fetchNext().then(
      (conversationList) => {
        let tempConversationList = this.state.conversationList;
        tempConversationList = [...this.state.conversationList,...conversationList];
        if(this.state.searchText && conversationList?.length == 30){
          this.fetchUserConversation();
        }
        this.setState({conversationList:tempConversationList, isLoading:false, isFlatListFooterLoading:false})
      },
      (error) => {
      
      }
    );
  }
  async fetchGroupConversation(){
    this.groupConversationsRequest.fetchNext().then(
      async (conversationList)  => {
        let tempConversationList = this.state.conversationList;
        tempConversationList = [...this.state.conversationList,...conversationList];
        if(this.state.searchText && conversationList?.length == 30){
          this.fetchGroupConversation();
        }
        this.setState({conversationList:tempConversationList, isLoading:false, isFlatListFooterLoading:false})
      },
      (error) => {
      }
    );
  }
  getGroupUnReadCount(){
    CometChat.getUnreadMessageCountForAllGroups().then((res:any)=>{
      let TotalUnReadCount = 0;
      for (const key in res) {
        if (res.hasOwnProperty(key)) {
        TotalUnReadCount = TotalUnReadCount + res[key];
        }
      }
      this.setState({unReadCount: TotalUnReadCount})
    })
  }
  getUserUnReadCount(){
    CometChat.getUnreadMessageCountForAllUsers().then((res:any)=>{
      let TotalUnReadCount = 0;
      for (const key in res) {
        if (res.hasOwnProperty(key)) {
        TotalUnReadCount = TotalUnReadCount + res[key];
        }
      }
      this.setState({unReadCount: TotalUnReadCount})
    })
  }
  getAllUnReadMessageCount(){
    if(this.state.screenFlag === 'groups'){
     this.getGroupUnReadCount();
    }else{
      this.getUserUnReadCount();
    }
  }
  onSearch(text: string){
    if(this.state.searchText === ''){
      this.getConversations();
    }
    this.setState({searchText: text})
  }
  
  setTagsOnCoversation(conversationID:string, conversationType: string,tags: Array<string>){
  CometChat.tagConversation(conversationID, conversationType, tags).then(
        conversation => {
            console.log('conversation', conversation);
        }, error => {
            console.log('error while fetching a conversation', error);
        }
    );
  }
  pinClick(item: any) {
    if(item?.conversationType=== 'group'){
      this.setTagsOnCoversation(item?.conversationWith?.guid,'group',['pin'])
    }else{
      this.setTagsOnCoversation(item?.conversationWith?.uid,'user',['pin'])
    }
  }
  DeleteClick(item: any) {
    this.setState({isLoading: true});
    let id;
    let type;
    if(item?.conversationType === 'group'){
      id = item?.conversationWith?.guid;
      type = 'group'

    }else{
      id = item?.conversationWith?.uid;
      type = 'user'
    }
    CometChat.deleteConversation(id, type).then(
      deletedConversation => {
          console.log(deletedConversation);
          this.setState({conversationList:[]})
          this.resetConversationRequest();
          this.getConversations();
      }, error => {
        this.setState({isLoading: false});
          console.log('error while deleting a conversation', error);
      }
  );
    
  }
  ArchiveClick(item: any) {
   
  }
  MoreClick(item: any) {
    
  }
  onPressItem(item: any){
    if(item?.conversationType === 'group'){
      this.props.navigation.navigate("ConversationScreen", {
        chatNumber: item?.conversationWith?.guid,
        mainConversationId: item?.conversationId,
        userName: item?.conversationWith?.name,
        userAvtar: item?.conversationWith?.avatar,
        userType: 2,
    })
    }else{
      this.props.navigation.navigate("ConversationScreen", {
        chatNumber: item?.conversationWith?.uid,
        mainConversationId: item?.conversationId,
        userName: item?.conversationWith.name,
        userAvtar: item?.conversationWith?.avatar,
        userType: 1,
    })
    }
  }
  onCreate(){
    if(this.state.screenFlag === 'groups'){
      this.props.navigation.navigate('UsersSelection',{title:"Select Participants ",multipleSelection:true});
    }else{
      this.props.navigation.navigate('UsersSelection',{title:"Select User",multipleSelection:false});
    }
  }
  onRefresh(){
    this.setState({conversationList:[],isLoading:true})
    this.getAllUnReadMessageCount();
    this.resetConversationRequest();
    this.getConversations();
  }
  endReached(){
    if(this.state.conversationList?.length >= 30 && !this.state.searchText){
      this.setState({isFlatListFooterLoading: true})
      this.getConversations()
    }
  }
  onBackPressOfUserSelection(){
    this.setState({isVisibleUserSelectionScreen:false})
  }
  onUserSelect(users:any){
    this.setState({isVisibleUserSelectionScreen:false})
  }
}
  
// Customizable Area End
