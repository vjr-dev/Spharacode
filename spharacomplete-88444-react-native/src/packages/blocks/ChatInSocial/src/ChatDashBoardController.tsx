// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { BlockComponent } from "framework/src/BlockComponent";
import { IBlock } from "framework/src/IBlock";
import { runEngine } from "framework/src/RunEngine";
import { GotoHomePage } from "../../../components/src/Navigation/NavigationFunctions";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { loginOnChat } from "../../../components/src/CometChatCommonFunctions";
import { Alert } from "react-native";

export const configJSON = require("./config");
export interface Props {
  navigation: any;
  id: string;
}
interface S {
  screenFlag:string,
  isUserLoggedIn: boolean
}
interface SS {
  id: any;
}

export default class ChatDashBoardController extends BlockComponent<Props, S, SS> {
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    this.state = {
        screenFlag:'chats',
        isUserLoggedIn: false
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }
  componentDidMount(): Promise<void> {
    this.getInitialData();
  }
  async getInitialData(){
    const tempUserData = await AsyncStorage.getItem("User_Data");
    const userData = JSON.parse(tempUserData);
    loginOnChat(userData.data.attributes?.full_phone_number).then((response:any)=>{
      this.setState({isUserLoggedIn: true})
    }).catch((error: any)=>{
      Alert.alert("","Something went wrong, please try again")
    })
  }
	onSwitch(screen:string){
		this.setState({screenFlag: screen})
	}
	onPressLogo(){
		GotoHomePage(this.props);
	}
}
// Customizable Area End
