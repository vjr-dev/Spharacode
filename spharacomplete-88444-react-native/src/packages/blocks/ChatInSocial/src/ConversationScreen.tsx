// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";
import {
  View,
  Text,
  SafeAreaView,
  Platform,
  StatusBar,
  ImageBackground,
  TouchableOpacity,
  Image,
} from "react-native";
import ConversationController, { Props } from "./ConversationController";
import { Styles } from "./ConversationStyle";
import * as IMAGE from "./assets";
import { COLORS } from "../../../framework/src/Globals";
import { GiftedChat } from "react-native-gifted-chat";
import { renderInputToolbar, renderComposer, renderActions, renderSend, renderBubble,renderLoading,rendermessageImage } from "../../../components/src/ChatViewFunctionality";

export default class ConversationScreen extends ConversationController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    const isMessageViewOnly = this.props.route.params?.isMessageViewOnly
    return (
      <SafeAreaView style={Styles.safeareaView}>
        {/* <View style={Styles.child}> */}
          <StatusBar
            animated={true}
            backgroundColor={isMessageViewOnly?"#f07135":"#454545"}
            barStyle={"light-content"}
            showHideTransition={"slide"}
            // hidden={true}
          />
          <ImageBackground source={IMAGE.back1} style={Styles.container}>
            <ImageBackground source={IMAGE.back2} style={Styles.container}>
             {
              isMessageViewOnly ? null :
              <View
              style={{
                backgroundColor: COLORS.darkorange,
                height: 63,
                alignItems: "center",
                flexDirection: "row",
              }}
            >
              <TouchableOpacity
                testID="upperBackButton"
                style={{
                  width: 60,
                  height: "100%",
                  justifyContent: "center",
                  alignItems: "center",
                }}
                onPress={() => this.upperBackPress()}
              >
                <Image
                  source={IMAGE.image_back}
                  style={{ height: 18, width: 18, tintColor: COLORS.white }}
                />
              </TouchableOpacity>
              <View
                style={{
                  height: 50,
                  width: 50,
                  borderRadius: 25,
                  justifyContent: "center",
                }}
              >
                {this.props.route.params?.userAvtar ? (
                  <Image
                    resizeMode="cover"
                    style={{
                      width: "100%",
                      height: "100%",
                      borderRadius: 25,
                    }}
                    source={{
                      uri: this.props.route.params.userAvtar,
                    }}
                  />
                ) : (
                  <View style={Styles.contactText}>
                    <Text
                      style={{
                        textTransform: "uppercase",
                        color: "black",
                        fontWeight: "bold",
                        fontSize: 18,
                      }}
                    >
                      {this.props.route.params?.userName?.charAt(0)}
                    </Text>
                  </View>
                )}
              </View>
              <Text
                style={{ color: COLORS.white, fontSize: 15, paddingLeft: 10 }}
              >
                {this.props.route.params?.userName}
              </Text>
            </View>
             }            
              <GiftedChat
                inverted={isMessageViewOnly ? false : true}
                testID="giftChat"
                scrollToBottom
                isCustomViewBottom
                bottomOffset={18}
                lightboxProps={{ useNativeDriver: true }}
                messages={this.state.messages}
                onInputTextChanged={(e) => this.setState({ onSendMessage: e })}
                text={this.state.onSendMessage}
                renderSend={(props) => renderSend(props)}
                showUserAvatar={false}
                minInputToolbarHeight={Platform.OS === "ios" ? 40 : 70}
                renderActions={(props) => renderActions(props)}
                renderInputToolbar={(props) => isMessageViewOnly ? <View></View> : renderInputToolbar(props, this.sendMessageCometChat, this.state.onSendMessage )}
                renderComposer={(props) => renderComposer(props, this.handleSelectFile,this.onPressCamera,this.state.imageLoading)}
               // renderCustomView = {(props) => renderCustomView(props, this.handleSelectFile)}
                renderBubble={(props) => renderBubble(props,this.state.imageLoading)}
                renderLoading={(props) =>renderLoading(props)}
                renderMessageImage={(props) => rendermessageImage(props, this.state.imageLoading)}
                user={{
                  _id: this.state.myUserId,
                }}
              />
            </ImageBackground>
          </ImageBackground>
        {/* </View> */}
      </SafeAreaView>
    );
  }
}
// Customizable Area End
