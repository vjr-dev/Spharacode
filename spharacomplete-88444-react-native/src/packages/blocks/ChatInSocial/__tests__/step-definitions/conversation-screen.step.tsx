//@ts-nocheck
//@ts-ignore
import { shallow, ShallowWrapper } from "enzyme";
import { defineFeature, loadFeature } from "jest-cucumber";
import * as helpers from "../../../../framework/src/Helpers";
import React from "react";
import { View } from "react-native";
import ConversationScreen from "../../src/ConversationScreen";

const screenProps = {
  navigation: {
    navigate: jest.fn(),
    pop: jest.fn(),
    goBack: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),
    getParam: jest.fn(),
    state: {
      params: {
        userAvtar: "xyz",
        chatNumber: 1,
        userType: 1,
        mainConversationId: "1234",
      },
    },
  },
  route: {
    params: {
      isMessageViewOnly: "",
      userAvtar: ""
    },
  },
  id: "Chat",
};

const screenProps2 = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),
    getParam: jest.fn(),
        state: {
      params: {
        userAvtar: "xyz",
        chatNumber: 1,
        userType: 2,
        mainConversationId: "1234",
        isMessageViewOnly: "",
      },
    },
  },
  id: "Chat",
};
const feature = loadFeature(
  "./__tests__/features/conversation-screen-scenario.feature"
);

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to ConversationScreen", ({ given, when, then }) => {
    let conversationWrapper: ShallowWrapper;
    let instance: ConversationScreen;

    given("I am a User loading ConversationScreen", () => {
      conversationWrapper = shallow(<ConversationScreen {...screenProps} />);
      expect(conversationWrapper).toBeTruthy();
      expect(conversationWrapper).toMatchSnapshot();
    });

    when("I navigate to the ConversationScreen", () => {
      instance = conversationWrapper.instance() as ConversationScreen;
    });

    then("I can press upper back button", () => {
      let upperBackButton = conversationWrapper.findWhere(
        (node) => node.prop("testID") === "upperBackButton"
      );
      upperBackButton.simulate("press");
      expect(upperBackButton).toBeTruthy();
    });

    then("Display conversation without error", () => {
      let giftChat = conversationWrapper.findWhere(
        (node) => node.prop("testID") === "giftChat"
      );
      giftChat.props().onInputTextChanged("aaa");
      giftChat.props().renderActions();
      giftChat.props().renderInputToolbar({ text: "bb" });
      giftChat.props().renderComposer({ text: "bb" });
      expect(giftChat).toBeTruthy();
    });

    then("I can send the message", () => {
      instance.componentWillUnmount();
      instance.onPageLeave();
      instance.receive();
      instance.listenForMessages();
      instance.transformSingleMessageForListen();
      instance.isValidMessage({ message: "1234" });
      instance.transformMessages([{ messages: 2 }]);
      instance.componentDidMount();
      instance.transformedSingleMessageObject();
      instance.CallMultiplefunctionToReduceComplexityTwo();
      instance.CallMultiplefunctionToReduceComplexityOne();
      instance.getFileName("abc.png", "photo");
      instance.getFileName("abc.png", "video");
      instance.getFileName("abc.png", "other");
      instance.handleSelectFile();
      instance.sendMediaMessageCometChat();
      // conversationWrapper = shallow(<ConversationScreen {...screenProps2} />);
      // instance = conversationWrapper.instance() as ConversationScreen;
      instance.loadMessages(2);
      instance.sendMessageCometChat(2);
    });
  });
});
