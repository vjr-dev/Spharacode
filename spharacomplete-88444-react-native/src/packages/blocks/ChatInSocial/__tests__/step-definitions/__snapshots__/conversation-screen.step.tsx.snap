// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conversation Screen User navigates to ConversationScreen 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
      "width": 750,
    }
  }
>
  <StatusBar
    animated={true}
    backgroundColor="#454545"
    barStyle="light-content"
    showHideTransition="slide"
  />
  <ImageBackground
    source={
      Object {
        "testUri": "../../../packages/blocks/ChatInSocial/assets/image_background-1.png",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "width": 750,
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/ChatInSocial/assets/image_background-2.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": "100%",
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255,138,73,1)",
            "flexDirection": "row",
            "height": 63,
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "width": 60,
            }
          }
          testID="upperBackButton"
        >
          <Image
            source={
              Object {
                "testUri": "../../../packages/blocks/ChatInSocial/assets/image_back.png",
              }
            }
            style={
              Object {
                "height": 18,
                "tintColor": "rgb(255,255,255)",
                "width": 18,
              }
            }
          />
        </ForwardRef>
        <View
          style={
            Object {
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255,211,154,1)",
                "borderRadius": 25,
                "height": 50,
                "justifyContent": "center",
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textTransform": "uppercase",
                }
              }
            />
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "fontSize": 15,
              "paddingLeft": 10,
            }
          }
        />
      </View>
      <GiftedChat
        bottomOffset={18}
        inverted={true}
        isCustomViewBottom={true}
        lightboxProps={
          Object {
            "useNativeDriver": true,
          }
        }
        messages={Array []}
        minInputToolbarHeight={70}
        onInputTextChanged={[Function]}
        renderActions={[Function]}
        renderBubble={[Function]}
        renderComposer={[Function]}
        renderInputToolbar={[Function]}
        renderLoading={[Function]}
        renderMessageImage={[Function]}
        renderSend={[Function]}
        scrollToBottom={true}
        showUserAvatar={false}
        testID="giftChat"
        text=""
        user={
          Object {
            "_id": "",
          }
        }
      />
    </ImageBackground>
  </ImageBackground>
</ForwardRef(SafeAreaView)>
`;
